File: app/src/main/java/gateway/hitrontech/com/encryption/fragment/encryption/Contract.java
package gateway.hitrontech.com.encryption.fragment.encryption;

import moe.xing.mvp_utils.BasePresenter;
import moe.xing.mvp_utils.BaseView;

interface Contract {

  interface Presenter extends BasePresenter {

    void getEncryptionText(String plainText);
  }

  interface View extends BaseView<Presenter> {

    void setEncryptionText(String cipherText);
  }
}

File: app/src/main/java/gateway/hitrontech/com/encryption/fragment/encryption/Contract.kt
package gateway.hitrontech.com.encryption.fragment.encryption

import moe.xing.mvp_utils.BasePresenter
import moe.xing.mvp_utils.BaseView

interface Contract {

    interface Presenter : BasePresenter {

        fun getEncryptionText(plainText: String)
    }

    interface View : BaseView<Presenter> {

        fun setEncryptionText(cipherText: String)
    }
}
